### PR TITLE
Add estimation of optimal number of training steps and sampling size for fully-visible GRBM

### PR DESCRIPTION
**Pull request by VolodyaCO**
This PR adds methods to the GRBM class that compute the required learning rate, data sample size, GRBM sample size and total number of optimisation steps that guarantee convergence of the GRBM, as explained in https://arxiv.org/abs/2306.14969. The GRBM prepares a Boltzmann distribution, so its state can be considered a thermal state of the longitudinal Ising Hamiltonian (see https://arxiv.org/abs/2512.03162 and https://arxiv.org/abs/2511.03796). Therefore, we can use the results of the first paper.

### CODE REVIEW

**Comment by VolodyaCO** [comment]
I forgot to turn my black formatter off again. Sorry. There are not too many changes to the formatting, though.

**Review by kevinchern — CHANGES_REQUESTED** [review]
Will take a little longer to give a full review--need to read/skim the references.

In the meantime...
1) Could you undo the formatting changes? Try to avoid formatting changes in general; formatting changes make it difficult to review code and, separately, breaks commit history e.g., "git blame". If there's an intentional and explicit formatting change, do it in a dedicated formatting-PR and then rebase the current PR on that formatting-PR. This will only fix the readability issue but not the git blame issue.
2) Can you add equation references in comment like [this example](https://github.com/dwavesystems/dimod/pull/1414/changes#diff-ae251a782bc8538320ec631f7d76ce575e2a337304cb64fbb39319cb04c1aba2R54-R63)?

**Comment by VolodyaCO** [comment]
Ok I think I was able to undo all the formatting changes and now finally disabled my formatter. Also added some references.
